.github/workflows/publish_storybook.yml

```diff
@@ -0,0 +1,59 @@
+# Workflow name
+name: Build and Publish Storybook to GitHub Pages
+
+on:
+  # Event for the workflow to run on
+  push:
+    branches:
+      - "main" # Replace with the branch you want to deploy from
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# List of jobs
+jobs:
+  retrieve_pages_url:
+    runs-on: ubuntu-latest
+    outputs:
+      pages_url: ${{ steps.get_url.outputs.url }}
+    steps:
+      - id: get_url
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          url=$(gh api "repos/$GITHUB_REPOSITORY/pages" --jq '.html_url')
+          echo "url=$url" >> "$GITHUB_OUTPUT"
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.build-publish.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: retrieve_pages_url
+    # Job steps
+    steps:
+      # Manual Checkout
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      # Install pnpm
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+      # Set up Node
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: "pnpm"
+          cache-dependency-path: "./pnpm-lock.yaml"
+      #👇 Add Storybook build and deploy to GitHub Pages as a step in the workflow
+      - uses: bitovi/github-actions-storybook-to-github-pages@v1.0.3
+        id: build-publish
+        env:
+          PAGES_URL: ${{ needs.retrieve_pages_url.outputs.pages_url }}
```

Comment:
you can just `"https://${GITHUB_REPOSITORY_OWNER}.github.io/${GITHUB_REPOSITORY#*/}/${{ github.event.inputs.branch || 'main' }}/"` but whatever

Reply:
I'll write it down as custom domain support that's completely unnecessary.